Add data to API

Infernia829 opened the pull request:
I added a list of connected players to the API, its confirmed working on a local server.

Attached below is a picture of the resulting .json.

![confirmed_working](https://github.com/MT-CTF/server_stats/assets/106501309/7f29dad8-5ebf-434f-b546-73728b903e83)

Infernia829 committed:
Add data to API

I added a list of connected players to the API, its confirmed working on a local server, will attach picture.

LoneWolfHT commented on init.lua:
```diff
@@ -9,6 +22,18 @@ if table.indexof(minetest.get_dir_list(worldpath, true), "api") == -1 then
 	minetest.mkdir(worldpath.."/api/")
 end
 
+local function get_player_list()
+	local player_names = {}
+	local players = minetest.get_connected_players()
+
+	for _, player in ipairs(players) do
+    	local name = player:get_player_name()
+    	table.insert(player_names, name)
+	end
+
+	return minetest.write_json(player_names), table.getn(player_names)
+end
```
```suggestion
local function get_player_list()
	local player_names = {}

	for _, player in ipairs(minetest.get_connected_players()) do
    	table.insert(player_names, player:get_player_name())
	end

	return player_names, #player_names
end
```

LoneWolfHT commented on init.lua:
```diff
@@ -1,6 +1,19 @@
+
+-- data cache --
+
+local name_mode = ''
+local matches_mode = ''
+local matches_played_mode = ''
+local name_map = ''
+local technical_name_map = ''
+local start_time_map = ''
```
```suggestion
local cache = {}
```

Infernia829 replied:
I thought there was a better way, just plumb slipped my mind.

LoneWolfHT commented on init.lua:
```diff
@@ -18,16 +43,59 @@ local function update(data)
 end
 
 ctf_api.register_on_new_match(function()
+
+	local player_names, player_count = get_player_list()
+
+	name_map = ctf_map.current_map.name
+	technical_name_map = ctf_map.current_map.dirname
+	start_time_map = os.time() -- Can be converted to local date here https://www.unixtimestamp.com/
+	name_mode = ctf_modebase.current_mode
+	matches_mode = ctf_modebase.current_mode_matches
+	matches_played_mode = ctf_modebase.current_mode_matches_played
+
```
```suggestion
```

LoneWolfHT replied:
Basically just revert the changes you made when adding these non-playerlist variables

LoneWolfHT reviewed (changes requested):
My suggestions will get you started on the proper implementation

Infernia829 committed:
Add changes

Add changes suggested by @LoneWolfHT.

Infernia829 commented:
Changes pushed, confirmed working: 
![image](https://github.com/MT-CTF/server_stats/assets/106501309/9000d6e0-86d9-4b02-b986-50e8b7c59046)

LoneWolfHT commented on init.lua:
```diff
@@ -18,16 +34,56 @@ local function update(data)
+    
+    if timer >= player_update_delay then
+		local player_names, player_count = get_player_list()
+        update({
+			current_mode = {
+				name = cache[0],
+				matches = cache[1],
+				matches_played = cache[2],
+			},
+			current_map = {
+				name = cache[3],
+				technical_name = cache[4],
+				start_time = cache[5], 
+			},
+			player_info = {
+				players = player_names,
+				count = player_count
+			},
+		})
```
```suggestion
        cache.player_info = {
             players = player_names,
             count = player_count
        }
        update(cache)
```

LoneWolfHT commented:
For some reason my fourth suggestion didn't appear, see my latest one on how to use the `cache` var

Infernia829 committed:
More efficient code

Updated the cache usage based off of @LoneWolfHT's suggestion, and stop using globalstep, in favor of leaveplayer/joinplayer.

Infernia829 commented:
Made changes based of of @LoneWolfHT's suggestions, and stopped using globalstep() in favor of joinplayer/leaveplayer.

Confirmed working:

![image](https://github.com/MT-CTF/server_stats/assets/106501309/53873d1f-b680-4ca4-9dad-f333d3f5ee29)
